[Feature]: Implemented User Authentication and Registration

hariketsheth opened the pull request:
## Description
This pull request addresses Issue #1 , which proposes the addition of User Authentication and Registration Functionality to the project. The following tasks are completed as a part of the same
- [X] Implemented Passport.js and LocalStrategy for Authentication
- [X] Integrated bcrypt for secure Password Hashing
- [X] Connected User Model to store user information
- [X] Added Update User Profile Feature
- [X] Added Validation to Register Route

## Related Issue
Fixes #1 

## Screenshots
![image](https://github.com/srummanf/Blog---MEN-Stack/assets/72455881/406bc6e9-e164-4064-a917-011b36a7562a)

![image](https://github.com/srummanf/Blog---MEN-Stack/assets/72455881/13e0a6cd-4c97-4a5a-ac95-37ed29f1be1d)


![image](https://github.com/srummanf/Blog---MEN-Stack/assets/72455881/d175f76b-8705-4052-9174-e45c7a7df2d2)


## Checklist
- [X] Code follows the project's coding guidelines
- [X] New code is adequately documented
- [X] No sensitive data is exposed in the logs or code
- [X] README and documentation have been updated (if necessary)
- [X] Ensuring No Merge Conflicts will happen upon merging the PR

@srummanf If you could please review this PR. 🎉
Thank you !

Commit:
Added Passport Configuration

Commit:
Added Registration Route

Commit:
Added Login Route

Commit:
Logout Route Added

srummanf commented:
Hello Bhaiyya, Can you check server>routes>admin.js  file, I have already added that functionality. You can make these changes at that file.

Commit:
Updated Routes and Registration

Commit:
Added User Profile Update Feature

hariketsheth commented:
> Hello Bhaiyya, Can you check server>routes>admin.js file, I have already added that functionality. You can make these changes at that file.

Hey Rumman,
I missed that file out. I have removed the redundant routes. 
Additionally, I have added `validation` to the register route and added `Profile Updation`
